apps/web/scripts/integrity-hashes.cjs

```diff
@@ -0,0 +1,65 @@
+const fs = require('fs')
+const path = require('path')
+const crypto = require('crypto')
+const cheerio = require('cheerio')
+
+const OUT_DIR = 'out'
+
+/**
+ * Given a local file path, compute the SHA-256 integrity hash.
+ */
+function generateIntegrityHash(filePath) {
+  const fileContents = fs.readFileSync(filePath)
+  const hash = crypto.createHash('sha256').update(fileContents).digest('base64')
+  return `sha256-${hash}`
+}
+
+/**
+ * Process a single .html file to add SRI attributes to local script tags.
+ */
+function processHtmlFile(htmlFilePath) {
+  const html = fs.readFileSync(htmlFilePath, 'utf8')
+  const $ = cheerio.load(html)
+
+  $('script[src]').each((_, scriptEl) => {
+    const scriptSrc = $(scriptEl).attr('src')
+    // Skip external or protocol-based (http/https) scripts:
+    if (!scriptSrc || scriptSrc.startsWith('http')) {
+      return
+    }
```

Comment:
We don't load any external scripts right now but if we wanted to support integrity hashes we would need to fetch the scripts here e.g. via curl.

Reply:
I added a comment and a log but I suggest to leave it out of scope for this PR.